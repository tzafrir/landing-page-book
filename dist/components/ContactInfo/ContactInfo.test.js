'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _ContactInfo = require('./ContactInfo');

var _ContactInfo2 = _interopRequireDefault(_ContactInfo);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('ContactInfo', function () {
  var props = void 0;
  var mounted = void 0;
  var contactInfo = function contactInfo() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_ContactInfo2.default, props));
    }
    return mounted;
  };
  describe('ContactInfo', function () {
    beforeEach(function () {
      props = {
        showBorder: true,
        showIcons: true,
        lng: 0,
        lat: 0,
        locationApiKey: '',
        address: 'Front10 address',
        email: 'hello@front10.com',
        phone: '55555555',
        mobile: '5555555',
        fax: '6554 6546 4664 655546',
        website: 'www.front10.com',
        contentAlign: 'left'
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_ContactInfo2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = contactInfo().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});