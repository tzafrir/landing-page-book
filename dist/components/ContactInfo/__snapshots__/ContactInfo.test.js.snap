// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo ContactInfo should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactInfo
    address="Front10 address"
    contentAlign="left"
    email="hello@front10.com"
    fax="6554 6546 4664 655546"
    lat={0}
    lng={0}
    locationApiKey=""
    mobile="5555555"
    phone="55555555"
    showBorder={true}
    showIcons={true}
    website="www.front10.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        0,
        <div
          className="ContactInfo__Information"
        >
          <Icon
            className=""
            icon="fa fa-map-marker"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
          Front10 address
        </div>,
        <div
          className="ContactInfo__Information"
        >
          <Icon
            className=""
            icon="fa fa-envelope"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
          hello@front10.com
        </div>,
        <div
          className="ContactInfo__Information"
        >
          <Icon
            className=""
            icon="fa fa-phone"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
          55555555
        </div>,
        <div
          className="ContactInfo__Information"
        >
          <Icon
            className=""
            icon="fa fa-mobile"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
          5555555
        </div>,
        <div
          className="ContactInfo__Information"
        >
          <Icon
            className=""
            icon="fa fa-fax"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
          6554 6546 4664 655546
        </div>,
        <div
          className="ContactInfo__Information"
        >
          <Icon
            className=""
            icon="fa fa-globe"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
          www.front10.com
        </div>,
      ],
      "contentAlign": "left",
      "image": "",
      "imageBorder": false,
      "imageCircle": true,
      "imageShadow": false,
      "shadow": false,
      "showBorder": true,
      "subTitle": "",
      "summary": "",
      "summaryJustified": false,
      "title": "",
    },
    "ref": null,
    "rendered": Array [
      0,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              className=""
              icon="fa fa-map-marker"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            "Front10 address",
          ],
          "className": "ContactInfo__Information",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-map-marker",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Front10 address",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              className=""
              icon="fa fa-envelope"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            "hello@front10.com",
          ],
          "className": "ContactInfo__Information",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-envelope",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "hello@front10.com",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              className=""
              icon="fa fa-phone"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            "55555555",
          ],
          "className": "ContactInfo__Information",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-phone",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "55555555",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              className=""
              icon="fa fa-mobile"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            "5555555",
          ],
          "className": "ContactInfo__Information",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-mobile",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "5555555",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              className=""
              icon="fa fa-fax"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            "6554 6546 4664 655546",
          ],
          "className": "ContactInfo__Information",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-fax",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "6554 6546 4664 655546",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              className=""
              icon="fa fa-globe"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            "www.front10.com",
          ],
          "className": "ContactInfo__Information",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-globe",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "www.front10.com",
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          0,
          <div
            className="ContactInfo__Information"
          >
            <Icon
              className=""
              icon="fa fa-map-marker"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
            Front10 address
          </div>,
          <div
            className="ContactInfo__Information"
          >
            <Icon
              className=""
              icon="fa fa-envelope"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
            hello@front10.com
          </div>,
          <div
            className="ContactInfo__Information"
          >
            <Icon
              className=""
              icon="fa fa-phone"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
            55555555
          </div>,
          <div
            className="ContactInfo__Information"
          >
            <Icon
              className=""
              icon="fa fa-mobile"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
            5555555
          </div>,
          <div
            className="ContactInfo__Information"
          >
            <Icon
              className=""
              icon="fa fa-fax"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
            6554 6546 4664 655546
          </div>,
          <div
            className="ContactInfo__Information"
          >
            <Icon
              className=""
              icon="fa fa-globe"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
            www.front10.com
          </div>,
        ],
        "contentAlign": "left",
        "image": "",
        "imageBorder": false,
        "imageCircle": true,
        "imageShadow": false,
        "shadow": false,
        "showBorder": true,
        "subTitle": "",
        "summary": "",
        "summaryJustified": false,
        "title": "",
      },
      "ref": null,
      "rendered": Array [
        0,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                className=""
                icon="fa fa-map-marker"
                onClick={[Function]}
                style={Object {}}
                title=""
              />,
              "Front10 address",
            ],
            "className": "ContactInfo__Information",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "icon": "fa fa-map-marker",
                "onClick": [Function],
                "style": Object {},
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Front10 address",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                className=""
                icon="fa fa-envelope"
                onClick={[Function]}
                style={Object {}}
                title=""
              />,
              "hello@front10.com",
            ],
            "className": "ContactInfo__Information",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "icon": "fa fa-envelope",
                "onClick": [Function],
                "style": Object {},
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "hello@front10.com",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                className=""
                icon="fa fa-phone"
                onClick={[Function]}
                style={Object {}}
                title=""
              />,
              "55555555",
            ],
            "className": "ContactInfo__Information",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "icon": "fa fa-phone",
                "onClick": [Function],
                "style": Object {},
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "55555555",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                className=""
                icon="fa fa-mobile"
                onClick={[Function]}
                style={Object {}}
                title=""
              />,
              "5555555",
            ],
            "className": "ContactInfo__Information",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "icon": "fa fa-mobile",
                "onClick": [Function],
                "style": Object {},
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "5555555",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                className=""
                icon="fa fa-fax"
                onClick={[Function]}
                style={Object {}}
                title=""
              />,
              "6554 6546 4664 655546",
            ],
            "className": "ContactInfo__Information",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "icon": "fa fa-fax",
                "onClick": [Function],
                "style": Object {},
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "6554 6546 4664 655546",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                className=""
                icon="fa fa-globe"
                onClick={[Function]}
                style={Object {}}
                title=""
              />,
              "www.front10.com",
            ],
            "className": "ContactInfo__Information",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "icon": "fa fa-globe",
                "onClick": [Function],
                "style": Object {},
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "www.front10.com",
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
